src/workerd/jsg/modules.h

```diff
@@ -175,7 +175,18 @@ class ModuleRegistry {
                v8::Local<v8::Module> module,
                kj::Maybe<SyntheticModuleInfo> maybeSynthetic = nullptr);
 
-    ModuleInfo(jsg::Lock& js, kj::StringPtr name, kj::ArrayPtr<const char> content);
+    enum class CompileFlags {
```

Comment:
Do we really need to make this an option? Why would a caller ever use `NONE` here rather than `EXTERNAL`?

Reply:
Yes. Worker bundles use this constructor and pass `NONE`. Also see https://github.com/cloudflare/workerd/pull/201#discussion_r1038656179, https://github.com/cloudflare/workerd/pull/201#discussion_r1038866995 and the experimental PR for compiler cache to see how else this would be used.

Reply:
I see, this is for built-ins.

I wonder if the flag should state that the module is a bulit-in, rather than trying to have each individual behavior be controllable?

Reply:
Yep, I'm already considering that. I wanted to run experiments separately (e.g. external strings no compile cache, heap strings with compile cache, external strings with compile cache) at first tho, so I can easily gauge the impact of each.

Reply:
For now, I'm adding a TODO comment here.